src/include/migraphx/tune_axis.hpp

```diff
@@ -24,21 +24,21 @@
 #ifndef MIGRAPHX_GUARD_OPERATORS_TUNE_AXIS_HPP
 #define MIGRAPHX_GUARD_OPERATORS_TUNE_AXIS_HPP
 
-#include <utility>
-#include <cstdint>
 #include <migraphx/stringutils.hpp>
 #include <migraphx/errors.hpp>
 
 namespace migraphx {
 inline namespace MIGRAPHX_INLINE_NS {
 
-inline int tune_axis(const int n_dim, const int axis, const std::string& op_name = "OPERATOR")
+inline int tune_axis(int n_dim, int axis, const std::string& op_name = "OPERATOR")
 {
-    if(axis >= n_dim or std::abs(axis) > n_dim)
-    {
+    if(axis < 0)
+        axis += n_dim;
+
+    if(axis < 0 or axis >= n_dim)
```

Comment:
Why this logic was changed ? I think previous way of just checking for `abs(axis)` is cleaner IMO

Reply:
It is logically confusing to do a comparison of `axis` twice in the same conditional -- once as its absolute version, the other as its original version. What is this code really doing by its intent..? And it is thus simpler to handle the negative axis first, and then do the right boundary checks. Thanks.